cdap-docs/admin-manual/source/security/authorization.rst

```diff
@@ -354,97 +347,102 @@ Kerberos Principal
-    one of ``namespace``, ``artifact``, ``application``, ``dataset``, ``program``, ``stream``, ``dataset_type`` or
-    ``dataset_module``.
+    > create role <role-name>
 
-  - For namespaces, ``<entity-id>`` is composed from the namespace, such as
-    ``namespace:<namespace-name>``.
+- To grant a role privileges to perform certain actions on an entity, use::
 
-  - For datasets, streams, artifacts and apps, ``<entity-id>`` is the namespace and entity names, such as
-    ``<namespace-name>.<dataset-name>``, ``<namespace-name>.<stream-name>``, ``<namespace-name>.<artifact-name>``,
-    and ``<namespace-name>.<app-name>``.
+    > grant actions <actions> on entity <entity-id> to role <role-name>
+    > revoke actions <actions> on entity <entity-id> from role <role-name>
 
-  - For programs, ``<entity-id>`` includes the application name and the program type:
-    ``<namespace-name>.<app-name>.<program-type>.<program-name>``. ``<program-type>`` is
-    one of flow, mapreduce, service, spark, worker, or workflow.
+  where:
 
-  - For datasets, streams, artifacts and apps, ``<entity-id>`` is the namespace and entity names, such as
-    ``<namespace-name>.<dataset-name>``, ``<namespace-name>.<stream-name>``, ``<namespace-name>.<artifact-name>``,
-    and ``<namespace-name>.<app-name>``.
+  - ``<actions>`` is a comma-separated list of privileges, any of *READ, WRITE, EXECUTE,* or *ADMIN*.
 
-  - ``<principal-type>`` can **only** be ``role`` since Sentry only supports granting privileges to roles.
+  - ``<entity>`` is of the form ``<entity-type>:<entity-id>``
 
-  - Wildcard can be used in each entity name to grant privileges to multiple entities. For example,
-    ``namespace:ns*`` represents all namespaces that starts with ``ns``.
-    ``namespace:ns?`` represents all namespaces that starts with ``ns`` and follows by a single character.
-    ``program:ns1.app1.*`` represents all types of programs in application app1 in namespace ns1.
+    =====================  =============================================
+        Entity Type                 Entity Id
+    =====================  =============================================
+    ``namespace``          ``<namespace>:<namespace-id>``
+    ``application``        ``<namespace-id>.<app-id>``
+    ``program``            ``<namespace-id>.<app-id>.<program-type>.<program-id>``
+    ``dataset``            ``<namespace-id>.<dataset-id>``
+    ``stream``             ``<namespace-id>.<stream-id>``
+    ``artifact``           ``<namespace-id>.<artifact-id>``
+    ``dataset-type``       ``<namespace-id>.<dataset-type-id>``
+    ``dataset_module``     ``<namespace-id>.<dataset-module-id>``
+    ``securekey``          ``<namespace-id>.<secure-key-id>``
+    ``kerberosprincipal``  ``<kerberos-principal-id>``
+    =====================  =============================================
 
-- To add the role to other principal, use::
+  - For programs, the program type is one of:
```

Comment:
For programs, there are two ways of granting wildcard privileges, one is `<namespace-id>.<app-id>.<program-type>.<program-id>`, the other is `<namespace-id>.<app-id>.*` to grant privileges to all program types in the application, I think it is better we mention it here.

Reply:
It is said in the next point